src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3980,6 +4000,83 @@ protected override RedisValue[] Parse(in RawResult result, out int count)
             }
         }
 
+        private static SortedSetRangeAndStoreCommandMessage CreateSortedSetRangeStoreMessage(
+            int db,
+            CommandFlags flags,
+            RedisKey destinationKey,
+            RedisKey sourceKey,
+            RedisValue start,
+            RedisValue stop,
+            SortedSetOrder sortedSetOrder,
+            Order order,
+            Exclude exclude,
+            long skip,
+            long take)
+        {
+            if (sortedSetOrder == SortedSetOrder.ByRank)
+            {
+                if (take > 0)
+                {
+                    throw new ArgumentException("take argument is not valid when sortedSetOrder is ByRank you may want to try setting the SortedSetOrder to ByLex or ByScore", nameof(take));
+                }
+
+                if (exclude != Exclude.None)
+                {
+                    throw new ArgumentException("exclude argument is not valid when sortedSetOrder is ByRank, you may want to try setting the sortedSetOrder to ByLex or ByScore", nameof(exclude));
+                }
+            }
+
+            var arguments = new List<RedisValue>();
+            arguments.Add(exclude == Exclude.Both || exclude == Exclude.Start ? $"({start}" : sortedSetOrder == SortedSetOrder.ByLex ? $"[{start}" : start);
+            arguments.Add(exclude == Exclude.Both || exclude == Exclude.Stop ? $"({stop}" : sortedSetOrder == SortedSetOrder.ByLex ?  $"[{stop}" : stop);
+
+            if (sortedSetOrder != SortedSetOrder.ByRank)
+            {
+                arguments.Add(sortedSetOrder.GetLiteral());
+            }
+
+            if (order == Order.Descending)
+            {
+                arguments.Add(RedisLiterals.REV);
+            }
+
+            if (take > 0)
+            {
+                arguments.Add(RedisLiterals.LIMIT);
+                arguments.Add(skip);
+                arguments.Add(take);
+            }
```

Comment:
I'm happy to take a pass at this but basically we want to avoid the `List` and array allocations here, likely by using an existing `Message` type. We may need to add a 7th on `CommandKeyValueValueValueValueValueMessage`. If you want, I can take the efficiency part of that on and push those bits :)

Reply:
I don't mind doing this, what's the correct pattern here?

I guess it would be `CommandKeyValueValueValueValueValueValueValueValueMessage`, set any unused value to null and just not write the nulls out?

Reply:
as there's the command, 2 keys, and at most 7 literals?